pkg/minikube/kubeconfig/kubeconfig.go

```diff
@@ -121,26 +119,11 @@ func UpdateEndpoint(contextName string, hostname string, port int, confpath stri
 		return false, errors.Wrap(err, "read")
 	}
 
-	address := "https://" + hostname + ":" + strconv.Itoa(port)
-
-	// if the kubeconfig is missed, create new one
```

Comment:
I think this might remove support for adding the context when it's been removed from `$HOME/.kube/config`

Do you mind checking that this works with your PR:

`minikube start`
`mv $HOME/.kube/config{,.bak}`
`minikube update-context`
`kubectl get po -A`
`mv $HOME/.kube/config{.bak,}`

Reply:
Yes, This PR removes support for adding the context when it's been removed from `$HOME/.kube/config`.

```
$ git show --quiet
commit c41a5f19b778ea6fb57a063022c9caceb11fe8ed (HEAD -> fix-nil-pointer-dereference, superbrothers/fix-nil-pointer-dereference)
Author: Kazuki Suda <kazuki.suda@gmail.com>
Date:   Thu Aug 13 12:19:42 2020 +0900

    minikube upate-cotext: Fix nil pointer dereference
$ out/minikube-darwin-amd64 version
minikube version: v1.12.3
commit: c41a5f19b778ea6fb57a063022c9caceb11fe8ed
$ mv $HOME/.kube/config{,.bak}
$ out/minikube-darwin-amd64 start --vm-driver virtualbox
😄  minikube v1.12.3 on Darwin 10.15.5
✨  Using the virtualbox driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating virtualbox VM (CPUs=2, Memory=6000MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.12 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube"
$ out/minikube-darwin-amd64 update-context

💣  update config: "minikube" does not appear in /Users/ksuda/.kube/config

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
$ mv $HOME/.kube/config{.bak,}
```

Originally, it was assumed that this command occurs an error if the cluster configuration is missed in the kubeconfig. Due to an implementation error in the test code, the following tests have not been ran so far (this is fixed at  a3fe0f5c499245802ee766f6e4a34349f657267c).

https://github.com/kubernetes/minikube/pull/8989/files#diff-289235936c30fdab30825ef3f90de144L298-L305

If we need to keep support for adding context, I would change the testcase above and the code as follows. What do you think?

```go
	// if the cluster setting is missed in the kubeconfig, create new one
	if _, ok := cfg.Clusters[contextName]; !ok {
		lp := localpath.Profile(contextName)
		gp := localpath.MiniPath()
		kcs := &Settings{
			ClusterName:          contextName,
			ClusterServerAddress: address,
			ClientCertificate:    path.Join(lp, "client.crt"),
			ClientKey:            path.Join(lp, "client.key"),
			CertificateAuthority: path.Join(gp, "ca.crt"),
			KeepContext:          false,
		}
		err = PopulateFromSettings(kcs, cfg)
		if err != nil {
			return false, errors.Wrap(err, "populating kubeconfig")
		}
	} else {
		cfg.Clusters[contextName].Server = address
	}
```

Reply:
We do need support for adding context. The test idea looks good. Thanks!

Reply:
I reverted support for adding context at https://github.com/kubernetes/minikube/pull/8989/commits/79d5641c5364c16dcc4d5e7cd64ceedfe78d04ab.

```
$ make out/minikube-darwin-amd64
GOOS="darwin" GOARCH="amd64" \
        go build -tags "container_image_ostree_stub containers_image_openpgp go_getter_nos3 go_getter_nogcs" -ldflags="-X k8s.io/minikube/pkg/version.version=v1.12.3 -X k8s.io/minikube/pkg/version.isoVersion=v1.12.2 -X k8s.io/minikube/pkg/version.isoPath=minikube/iso -X k8s.io/minikube/pkg/version.gitCommitID="eb01a2ef73dfde3c3a994b6719ff006b21d5822e" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v2" -a -o out/minikube-darwin-amd64 k8s.io/minikube/cmd/minikube
$ alias minikube="${PWD}/out/minikube-darwin-amd64"
$ git show --quiet
commit eb01a2ef73dfde3c3a994b6719ff006b21d5822e (HEAD -> fix-nil-pointer-dereference, superbrothers/fix-nil-pointer-dereference)
Author: Kazuki Suda <kazuki.suda@gmail.com>
Date:   Sun Aug 16 14:03:41 2020 +0900

    Update an integration test
$ minikube version
minikube version: v1.12.3
commit: eb01a2ef73dfde3c3a994b6719ff006b21d5822e
$ minikube delete
🔥  Deleting "minikube" in virtualbox ...
💀  Removed all traces of the "minikube" cluster.
$ minikube start
😄  minikube v1.12.3 on Darwin 10.15.5
✨  Using the virtualbox driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating virtualbox VM (CPUs=2, Memory=6000MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.12 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube"
$ mv $HOME/.kube/config{,.bak}
$ minikube update-context
🎉  "minikube" context has been updated to point to 192.168.99.135:8443
$ kubectl get po -A
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
kube-system   coredns-66bff467f8-st8dt           0/1     Running   0          18s
kube-system   etcd-minikube                      1/1     Running   0          23s
kube-system   kube-apiserver-minikube            1/1     Running   0          23s
kube-system   kube-controller-manager-minikube   1/1     Running   0          23s
kube-system   kube-proxy-qcllm                   1/1     Running   0          18s
kube-system   kube-scheduler-minikube            1/1     Running   0          22s
kube-system   storage-provisioner                1/1     Running   0          23s
$ mv $HOME/.kube/config{.bak,}
mv: overwrite '/Users/ksuda/.kube/config'? y
```